Rakefile

```diff
@@ -33,15 +35,28 @@ task :build_parser do
   `ragel -R lib/graphql/language/lexer.rl`
 end
 
+desc "Test the generated HTML files"
+task :html_proofer do
+  require "html-proofer"
+  `bundle exec nanoc compile`
+  HTMLProofer.check_directory("./site/output").run
+end
+
 desc "Build the site, copy it to the gh-pages branch, and push the gh-pages branch"
 task :deploy_site do
   # TODO: use master branch instead of site
-  `git checkout gh-pages` &&
-    `git checkout site -- site/ Gemfile` &&
-    (Dir.chdir("site") { `nanoc` }) &&
-    `cp -r site/output/graphql-ruby/ ./` &&
-    `git add -A` &&
-    `git commit -m "deploy site to gh-pages (automatic)"` &&
-    `git push origin gh-pages` &&
-    `git checkout site`
+  Dir.chdir("site") do
+    `bundle exec nanoc compile`
+    Dir.mktmpdir do |tmp|
+      system "mv output/* #{tmp}"
+      system "git checkout gh-pages"
+      system "rm -rf *"
+      system "mv #{tmp}/* ."
+      system "git add ."
+      system "git commit -am 'deploy site to gh-pages (automatic)'"
+      system "git push origin gh-pages --force"
+      system "git checkout master"
+      system "echo yolo"
```

Comment:
Ah, whoops. I almost always use `system` to be clear of the intent. First line was copy-pasted from somewhere else. 😅